### PR TITLE
feat(portalnet): add rfn lookup to gossip

### PR DESCRIPTION
**Pull request by njgheorghita**
### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: portalnet/src/gossip.rs [hunk 1]
```diff
@@ -40,38 +40,51 @@ pub struct GossipResult {
...
-                .filter(|node| node.status.is_connected())
-                .collect::<Vec<&kbucket::Node<NodeId, Node>>>()
-        })
-        .collect();
-
-    if all_nodes.is_empty() {
-        // If there are no nodes whatsoever in the routing table the gossip cannot proceed.
-        warn!("No nodes in routing table, gossip cannot proceed.");
-        return 0;
-    }
```
Comment (KolbyML) [inline portalnet/src/gossip.rs:67]: why was this removed?

Comment (KolbyML) [inline portalnet/src/gossip.rs:67]: Shouldn't this still be here?

**Comment by KolbyML** [comment]
I think the same result would have happened if you ran master on your home node running latest. So I think both should cases should have been tested as a control. 

Debugging and fixing why we are having provider issues is the highest issue then this. I do think this is a smarter way of gossipping.

Without the logs or seeing if there was provider errors I don't think I can comment anything of value on the results you told me of the tested you reported to me earlier today

**Comment by njgheorghita** [comment]
closing since this feature is no longer a priority, now that we've added direct offer to the state bridge
